### PR TITLE
Rchain 2841 test_successful_genesis_ceremony_with_read_only

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
Implement test_successful_genesis_ceremony_with_read_only

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-2841

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by zsluedem** [comment]
bors try

**Comment by ArturGajowy** [comment]
bors delegate+

File: integration-tests/test/wait.py [hunk 1]
```diff
@@ -140,7 +141,7 @@ def is_satisfied(self) -> bool:
         return actual_blocks_count >= self.blocks_count
 
 
-def wait_on_using_wall_clock_time(predicate: PredicateProtocol, timeout: int) -> None:
+def wait_on_using_wall_clock_time(predicate: PredicateProtocol, timeout: int, is_raise: bool= False) -> None:
```
Comment (adaszko) [inline integration-tests/test/wait.py]:
```suggestion
def wait_using_wall_clock_time_or_fail(predicate: PredicateProtocol, timeout: int) -> None:
```

File: integration-tests/test/wait.py [hunk 2]
```diff
@@ -162,9 +163,11 @@ def wait_on_using_wall_clock_time(predicate: PredicateProtocol, timeout: int) ->
 
         time.sleep(iteration_duration)
         elapsed = elapsed + iteration_duration
-
     logging.info("TIMEOUT %s", predicate)
-    pytest.fail('Failed to satisfy {} after {}s'.format(predicate, elapsed))
+    if is_raise:
+        raise WaitTimeoutError()
+    else:
+        pytest.fail('Failed to satisfy {} after {}s'.format(predicate, elapsed))
```
Comment (adaszko) [inline integration-tests/test/wait.py]:
```suggestion
        raise WaitTimeoutError(predicate, timeout)
```

File: integration-tests/test/wait.py [hunk 3]
```diff
@@ -162,9 +163,11 @@ def wait_on_using_wall_clock_time(predicate: PredicateProtocol, timeout: int) ->
...
```
Comment (adaszko) [inline integration-tests/test/wait.py]:
```suggestion
def wait_using_wall_clock_time(predicate, timeout):
    try:
        wait_using_wall_clock_time_or_fail(predicate, timeout)
    except TimeoutError:
        pytest.fail('Failed to satisfy {} after {}s'.format(predicate, elapsed))
```

File: integration-tests/test/common.py [hunk 4]
```diff
@@ -56,6 +56,9 @@ def __init__(self, command: Tuple[str, ...], exit_code: int, output: str) -> Non
 class GetBlockError(NonZeroExitCodeError):
     pass
 
+class WaitTimeoutError(Exception):
+    pass
```
Comment (adaszko) [inline integration-tests/test/common.py]:
```suggestion
    def __init__(self, predicate: PredicateProtocol, timeout: int) -> None:
        self.predicate = predicate
        self.timeout = timeout
```

**Comment by KentShikama** [comment]
bors try

**Comment by KentShikama** [comment]
@zsluedem Is this expected?

```
>                                   wait_for_approved_block_received_handler_state(context, readonly_a)
E                                   Failed: DID NOT RAISE <class 'test.common.WaitTimeoutError'>

test/test_genesis_ceremony.py:137: Failed
```

https://drone.rchain-dev.tk/rchain/rchain/1102/6

**Comment by KentShikama** [comment]
bors try

**Comment by zsluedem** [comment]
@KentShikama @adaszko 
https://github.com/rchain/rchain/blob/dev/docs/features.md#test-testtest_genesis_ceremonypytest_successful_genesis_ceremony_with_read_only
The feature.md says "readOnlyA never transitions to **ApprovedBlockReceivedHandler`**" but the readonly node did receive the `Valid ApprovedBlock` and `Making a transition to ApprovedBlockRecievedHandler state`. 
Are you guys sure that's what should be expected?

**Comment by KentShikama** [comment]
Yeah the read only node should indeed receive the approved block. Please change the document in this PR too.

**Comment by zsluedem** [comment]
@KentShikama So the readonly node can also `Making a transition to ApprovedBlockRecievedHandler state`, can't it? If so ,then this test is not really neccessary.

**Comment by KentShikama** [comment]
Well it shows how the read only node doesn't participate in the genesis ceremony. It is an relatively obvious/simple test. I don't see the harm in having it.

**Comment by zsluedem** [comment]
bors try

File: docs/features.md [hunk 5]
```diff
@@ -129,7 +129,7 @@
 * `ceremonyMaster` sends `ApprovedBlock` to `validatorA` and `validatorB`
 * `validatorA` and `validatorB` transition to ApprovedBlockReceivedHandler
 * `ceremonyMaster`, `validatorA` and `validatorB` tip points to block (genesis) where it has no parent and Bonds holds `validatorA` and `validatorB`
-* `readOnlyA` **never** transitions to `ApprovedBlockReceivedHandler`
+* `readOnlyA` transitions to `ApprovedBlockReceivedHandler`
```
Comment (adaszko) [inline docs/features.md]: @rabbitonweb ^

Comment (KellyatPyrofex) [reply docs/features.md]: This is the correct outcome. readOnlyA should not be able to validate the unapproved block AND should be able to receive the approved block and transition once the network launches.

**Comment by zsluedem** [comment]
@adaszko How can I merge this pr?

**Comment by KentShikama** [comment]
You need 2 approves now @zsluedem

**Pull request merged**